lib/liquid/extensions.rb

```diff
@@ -25,6 +26,12 @@ def to_liquid
   end
 end
 
+class BigDecimal
+  def to_liquid
+    to_f
```

Comment:
Well the same problem would be present if `to_s` was called on the input.  E.g. 

```ruby
irb> Liquid::Template.parse('{{ decimal }}').render("decimal" => BigDecimal.new("12.34"))
=> "0.1234E2"
```

so having the filter convert to a float just makes things less consistent.  I suppose this change makes things more consistent, but it doesn't help for preserving precision.